opendatahub-io/odh-dashboard
# Stop fetching model serving resources on details page when the feature is disabled

Close #1089  

## Description


Create a component named `ModelServingProjectDetailsContextAdditions`, which wraps the `ProjectDetailsContextProvider` and fetches the model serving resources there, and passes them to the provider. In the router, we check the feature flag and decide which component we should render.

## How Has This Been Tested?



1. Go to the project details page
2. Go to the network tab of the dev console of your browser (Chrome or Firefox)
3. Check the API calls to see if it's fetch `servingruntimes` and `inferenceservices`
4. Go to the dashboard config to disable the model serving feature
5. Wait for ~2 mins until the dashboard fetch the last config
6. Refresh the page to see the API calls, now the page shouldn't fetch `servingruntimes` and `inferenceservices`
7. Enable the model serving feature again
8. Check if the resources are fetched again

## Test Impact


Not applicable, cannot check what API calls are made and what are not for now.

## Request review criteria:
